compliance/README.md

```diff
@@ -0,0 +1,227 @@
+
+- [Audience](#audience)
+- [Background](#background)
+- [Architecture](#architecture)
+- [Endpoints](#endpoints)
+- [Schema](#schema)
+- [Examples](#examples)
+
+<a name="audience"></a>
+
+## Audience
+
+This specification has three intended audiences:
+
+1. Agencies, as consumers in the form of automated compliance measurement from their digitally-enforcable Policy documents
+2. Providers, so that they can get real-time feedback their compliance status as measured by their supervising Agency
+3. Tooling companies, who might want to create applications for visualization, analysis, and reporting on compliance status
+
+<a name="background"></a>
+
+## Background
+
+The [Policy API](../policy/readme.md) describes a schema for expressing mobility policy, as well as applicable geographic regions, in JSON document form. The Compliance API describes how Policy and Geography documents are combined with fleet data ingested via Agency to measure the degree to which Providers are in compliance with the policies.
+
+### Provider Compliance
+
+Many Policies require no global information, only vehicle status and telemetry that is available to a provider (e.g. speed limits can be measured on a per-vehicle basis). Having providers compute their own compliance status for these cases is desirable so as to prevent round-trip API calls to the city. However, the city must also compute compliance, for enforcement purposes. If the provider does not wish to compute this on their own, they can rely entirely on the city's computation.
+
+### Global Compliance
+
+Certain types of policy will be global across providerrs, e.g., instead of a per-provider fleet cap, a city might stipulate "the sum total of all permitted devices downtown may not exceed 2,000". This type of policy compliance cannot be computed without access to (anonymized) global knowledge. The compliance API will be R to provide whatever global state is needed, and the specific Policy will provide the paths to such endpoint(s). See the Policy API [specification](../policy/README.md) and [examples](../policy/README.md) of such policies.
+
+## Architecture
+
+The Compliance API takes as inputs the event and telemetry stream from the [Agency API](../agency/README.md), plus the published and applicable Policy documents from the [Policy API](../policy/README.md), and emits a JSON document that describes deviations from the various Policy documents.
+
+(Diagram to be added)
+
+## Endpoints
+
+Note: If a request comes from a Provider, the provider_id will be passed in the JWT authentication. If the request comes from an Agency, its token will not contain a provider_id and all Providers will be measured.
+
+`GET /snapshot/:policy_id`
+
+Parameters:
+
+| Name          | Type | R/O | Description                   |
+| ------------- | ---- | --- | ----------------------------- |
+| `provider_id` | UUID | O   | If not provided in the JWT.   |
+| `end_date`    | UUID | O   | Take snapshot at a given time |
```

Comment:
Hi, I guess the `UUID` type is a typo, and it's a timestamp in milliseconds as usual.

Reply:
It was indeed -- the spec has been updated.